### PR TITLE
[dvsim] Add sim_cfg.hjson files for all supported IPs

### PR DESCRIPTION
**Pull request by udinator**
Signed-off-by: Udi Jonnalagadda <udij@google.com>

### CODE REVIEW

File: hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      name: rv_dm_csr_aliasing
+      uvm_test_seq: rv_dm_csr_vseq
+      run_opts: ["+csr_aliasing", "+en_scb=0"]
+    }
+
+    {
+      name: rv_dm_mem_walk
+      uvm_test_seq: rv_dm_csr_vseq
+      run_opts: ["+csr_mem_walk", "+en_scb=0"]
+    }
```
Comment (sriyerg) [inline hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson]: These can be removed.

File: hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  testplan: "{proj_root}/hw/ip/rv_dm/data/rv_dm_testplan.hjson"
+
+  // RAL spec - used to generate the RAL model.
+  ral_spec: "{proj_root}/hw/ip/rv_dm/data/rv_dm.hjson"
+
+  // Import additional common sim cfg files.
+  import_cfgs: [// Project wide common sim cfg file
+                "{proj_root}/hw/dv/data/common_sim_cfg.hjson",
+                // Common CIP test lists
+                "{proj_root}/hw/dv/data/tests/csr_tests.hjson",
```
Comment (sriyerg) [inline hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson]: Include mem_tests.hjson here.

File: hw/ip/gpio/dv/gpio_sim_cfg.hjson [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      name: gpio_stress_all
+      uvm_test_seq: gpio_stress_all_vseq
+      run_opts: ["+do_clear_all_interrupts=0"]
+    }
+
+    {
+      name: gpio_csr_rw
+      run_opts: ["+do_clear_all_interrupts=0"]
+    }
+    {
```
Comment (sriyerg) [inline hw/ip/gpio/dv/gpio_sim_cfg.hjson]: nit: add newline

File: hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson (hunk 2 above)
Comment (sriyerg) [reply hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson]: Also check which other IPs aha

File: hw/ip/rv_timer/dv/rv_timer_sim_cfg.hjson [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      name: rv_timer_random_reset
+      uvm_test_seq: rv_timer_random_reset_vseq
+    }
+
+    {
+      name: rv_timer_stress_all
+      uvm_test_seq: rv_timer_stress_all_vseq
+      run_opts: ["+test_timeout_ns=10_000_000_000"]
+    }
+  ]
```
Comment (cindychip) [inline hw/ip/rv_timer/dv/rv_timer_sim_cfg.hjson]: Are these two tests included in the common tests already? I think we can remove them here.

File: hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson (hunk 1 above)
Comment (udinator) [reply hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson]: Done.

File: hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson (hunk 2 above)
Comment (udinator) [reply hw/ip/rv_dm/dv/rv_dm_sim_cfg.hjson]: Done.

File: hw/ip/aes/dv/aes_sim_cfg.hjson [hunk 5]
```diff
@@ -0,0 +1,51 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  // Name of the sim cfg - typically same as the IP under test.
```
[the remark this answers is not in the document]
Comment (udinator) [reply hw/ip/aes/dv/aes_sim_cfg.hjson]: Ack. Will change to "typically same as the name of the DUT", does this sound good?

File: hw/ip/gpio/dv/gpio_sim_cfg.hjson (hunk 3 above)
Comment (udinator) [reply hw/ip/gpio/dv/gpio_sim_cfg.hjson]: Good catch, updated.

File: hw/ip/rv_timer/dv/rv_timer_sim_cfg.hjson (hunk 4 above)
Comment (udinator) [reply hw/ip/rv_timer/dv/rv_timer_sim_cfg.hjson]: Good catch, it looks like the *_stress_all test is included in stress_tests.hjson, but the rv_timer_random_reset test is not included in the imported sim_cfg files.

**Comment by sriyerg** [comment]
Hey @udinator, 

Please update and rebase to get additional updates. Please add these new cfgs (except for rv_dm) to `hw/dv/data/master_sim_cfgs_list.hjson` file and run the following command:
```console
$ util/dvsim.py hw/dv/data/master_sim_cfgs_list.hjson -i all -n --map-full-testplan 
```
This will run ALL tests in all IPs in one shot in 'dry run' mode (due to the `-n` switch). It will not kick off the sims, it will only print the commands that would have executed. It is useful to do sanity checks in the hjson files in case there are typos. It will generate the testplan annotated results at the end.

File: hw/ip/spi_device/dv/spi_device_sim_cfg.hjson [hunk 6]
```diff
@@ -0,0 +1,66 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{
+  // Name of the sim cfg - typically same as the IP under test.
```
Comment (sriyerg) [inline hw/ip/spi_device/dv/spi_device_sim_cfg.hjson]: nit: same as the name of the DUT.

File: hw/ip/aes/dv/aes_sim_cfg.hjson (hunk 5 above)
[the remark this answers is not in the document]
Comment (sriyerg) [reply hw/ip/aes/dv/aes_sim_cfg.hjson]: Sure, please update UART and HMAC as well.

File: hw/ip/aes/dv/aes_sim_cfg.hjson [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  name: aes
+
+  // Top level dut name (sv module).
+  dut: aes
+
+  // Top level testbench name (sv module).
+  tb: tb
+
+  // Fusesoc core file used for building the file list.
+  fusesoc_core: lowrisc:dv:aes_sim:0.1
```
Comment (rasmus-madsen) [inline hw/ip/aes/dv/aes_sim_cfg.hjson]: this might change to 0.5 before the merge.. I will keep an eye out

**Comment by udinator** [comment]
> Hey @udinator,
> 
> Please update and rebase to get additional updates. Please add these new cfgs (except for rv_dm) to `hw/dv/data/master_sim_cfgs_list.hjson` file and run the following command:
> 
> ```
> $ util/dvsim.py hw/dv/data/master_sim_cfgs_list.hjson -i all_once -n --map-full-testplan 
> ```
> 
> This will run ALL tests in all IPs in one shot in 'dry run' mode (due to the `-n` switch). It will not kick off the sims, it will only print the commands that would have executed. It is useful to do sanity checks in the hjson files in case there are typos. It will generate the testplan annotated results at the end.

@sriyerg - thanks for the suggestion! I ran this command and things look good to me.

File: hw/ip/spi_device/dv/spi_device_sim_cfg.hjson (hunk 6 above)
Comment (udinator) [reply hw/ip/spi_device/dv/spi_device_sim_cfg.hjson]: Done.

File: hw/ip/uart/dv/uart_sim_cfg.hjson [hunk 8]
```diff
@@ -2,7 +2,7 @@
 // Licensed under the Apache License, Version 2.0, see LICENSE for details.
 // SPDX-License-Identifier: Apache-2.0
 {
-  // Name of the sim cfg - typically same as the DUT under test.
```
Comment (sriyerg) [inline hw/ip/uart/dv/uart_sim_cfg.hjson]: Also please update the template at `util/uvmdvgen/sim_cfg.hjson.tpl`

Comment (udinator) [reply hw/ip/uart/dv/uart_sim_cfg.hjson]: Done, thanks for directing me to this file.

**Pull request merged**